### PR TITLE
[EP-918] Added iOS theming

### PR DESCRIPTION
**Pull request by AndreiGalkin**

### CODE REVIEW

File: ios/Extensions/UIColor+Additions.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//
+//  UIColor+Additions.h
+//  RNJudo
+//
+//  Created by Andrei Galkin on 4/22/20.
+//  Copyright © 2020 Facebook. All rights reserved.
```
Comment (mpetrenco) [inline ios/Extensions/UIColor+Additions.h:6]: Facebook 😮.
Consider using the Judo headline like the other Obj-C wrapper classes.

File: ios/Wrappers/RNWrappers.m [hunk 2]
```diff
@@ -187,26 +187,65 @@ + (JPUIConfiguration *)uiConfigurationFromConfiguration:(NSDictionary *)configur
...
+                                        size:[[dictionary numberForKey:@"captionBoldSize"] doubleValue]];
+
+    theme.jpBlackColor = [UIColor colorFromHexString:[dictionary stringForKey:@"jpBlackColor"]];
+    theme.jpDarkGrayColor = [UIColor colorFromHexString:[dictionary stringForKey:@"jpDarkGrayColor"]];
+    theme.jpGrayColor = [UIColor colorFromHexString:[dictionary stringForKey:@"jpGrayColor"]];
+    theme.jpLightGrayColor = [UIColor colorFromHexString:[dictionary stringForKey:@"jpLightGrayColor"]];
+    theme.jpRedColor = [UIColor colorFromHexString:[dictionary stringForKey:@"jpRedColor"]];
+    theme.jpWhiteColor = [UIColor colorFromHexString:[dictionary stringForKey:@"jpWhiteColor"]];
+    theme.buttonColor = [UIColor colorFromHexString:[dictionary stringForKey:@"buttonColor"]];
+    theme.buttonTitleColor = [UIColor colorFromHexString:[dictionary stringForKey:@"buttonTitleColor"]];
```
Comment (mpetrenco) [inline ios/Wrappers/RNWrappers.m:247]: What about `button corner radius` and `back button custom image` ?

File: sample-app/screens/Home/HomeData.tsx [hunk 3]
```diff
@@ -128,7 +128,7 @@ const parsePaymentMethods = (values: Array<String>): JudoPaymentMethod => {
   return paymentMethods
 }
 
-const parseCardNetworks = (values: Array<String>): JudoCardNetwork => {
+const parseCardNetworks = (values: [string]): JudoCardNetwork => {
```
Comment (mpetrenco) [inline sample-app/screens/Home/HomeData.tsx:131]: isn't it supposed to be `string[]`?
Setting `[string]` means having an array with only 1 string element inside.

File: ios/Extensions/UIColor+Additions.h (hunk 1 above)
Comment (AndreiGalkin) [reply ios/Extensions/UIColor+Additions.h:6]: I don't even know why it is autogenerated with Facebook!

File: ios/Wrappers/RNWrappers.m (hunk 2 above)
Comment (AndreiGalkin) [reply ios/Wrappers/RNWrappers.m:247]: Will add them now

File: sample-app/screens/Home/HomeData.tsx (hunk 3 above)
Comment (AndreiGalkin) [reply sample-app/screens/Home/HomeData.tsx:131]: 👍

**Comment by mpetrenco** [comment]
One question, how do I set the system font ?

**Comment by AndreiGalkin** [comment]
@mpetrenco If you don't change font - it will be the same as in SDK for default

File: ios/Extensions/NSException+JudoValidationExceptions.h [hunk 4]
```diff
@@ -31,6 +31,7 @@ NS_ASSUME_NONNULL_BEGIN
 + (nonnull NSException *)exceptionUndefinedConfigurationWithKey:(nonnull NSString *)key;
 + (nonnull NSException *)exceptionNilConfigurationWithKey:(nonnull NSString *)key;
 + (nonnull NSException *)exceptionUnexpectedClassOfConfigurationWithKey:(nonnull NSString *)key andClass:(Class)aClass;
++ (NSException *)exceptionUnexpectedFormatConfigurationWithKey:(nonnull NSString *)key expectedFormat:(nonnull NSString *)format;
```
Comment (stpopa) [inline ios/Extensions/NSException+JudoValidationExceptions.h:34]: can we please `+ (nonnull NSException *)` instead of `+ (NSException *)`

**Pull request merged**